tests/httptoolkit/service/test_async_service.py

```diff
@@ -0,0 +1,242 @@
+        url="https://kontur.ru" + path + "?foo=bar",
+        text=response_text,
+    )
+
+    response = await async_service.post(
+        path,
+        headers=(Header(name="RequestHeader", value="request-header", is_sensitive=False),),
+        params={"foo": "bar"},
+        json={"param1": 1, "param2": 2},
+    )
+
+    assert response.status_code == 200
+    assert response.elapsed > timedelta(0)
+    assert response.text == response_text
+
+    calls = httpx_mock.get_requests()
+
+    assert calls[0].headers["RequestHeader"] == "request-header"
+    assert calls[0].headers["ServiceHeader"] == "async_service-header"
+    assert calls[0].headers["Content-Type"] == "application/json"
+    assert calls[0].content == b'{"param1": 1, "param2": 2}'
+
+
+@pytest.mark.asyncio
+async def test_if_response_not_successful(async_service: AsyncService, httpx_mock: HTTPXMock):
+    response_text = "Authorize first"
+    path = "/some/path"
+
+    httpx_mock.add_response(
+        method="GET",
+        url="https://kontur.ru" + path + "?foo=bar",
+        text=response_text,
+        status_code=401,
+    )
+
+    with pytest.raises(HttpError) as error:
+        await async_service.get(
+            path,
+            headers=(Header(name="RequestHeader", value="request-header", is_sensitive=False),),
+            params={"foo": "bar"},
+        )
+
+    assert "Authorize first" in str(error.value)
+    assert error.value.response_code() == 401
+
+    calls = httpx_mock.get_requests()
+
+    assert calls[0].headers["ServiceHeader"] == "async_service-header"
+    assert calls[0].headers["RequestHeader"] == "request-header"
+
+
+@pytest.mark.asyncio
+async def test_post_when_transport_error(async_service: AsyncService, httpx_mock: HTTPXMock):
+    path = "/some/path"
+
+    httpx_mock.add_exception(
+        method="POST",
+        url="https://kontur.ru" + path + "?foo=bar",
+        exception=httpx.TimeoutException("Timeout reached"),
+    )
+
+    with pytest.raises(ServiceError) as error:
+        await async_service.post(
+            path,
+            headers=(Header(name="RequestHeader", value="request-header", is_sensitive=False),),
+            params={"foo": "bar"},
+            body="body",
+        )
+
+    assert "TimeoutException: Timeout reached" in str(error.value)
+
+    calls = httpx_mock.get_requests()
+
+    assert calls[0].headers["ServiceHeader"] == "async_service-header"
+    assert calls[0].headers["RequestHeader"] == "request-header"
+    assert calls[0].content == b"body"
+
+
+def test_headers(async_service_with_only_url):
+    assert async_service_with_only_url.headers == ()
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("method", ("post", "put", "delete", "patch"))
+async def test_json_request_with_falsy_values(
+    async_service: AsyncService,
+    httpx_mock: HTTPXMock,
+    method: str,
+) -> None:
+    httpx_mock.add_response(
+        method=method.upper(),
+        url="https://kontur.ru/",
```

Comment:
It's better to replace real domain examples with "example.com" or similar.

Reply:
fixed